src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -348,14 +348,20 @@ class CommonVisitor : public AST::BaseVisitor<Derived> {
                 } else {
                     this->visit_expr(*s->m_slice);
                     ASR::expr_t *value = ASRUtils::EXPR(tmp);
-                    if (!ASR::is_a<ASR::ConstantInteger_t>(*value)) {
-                        throw SemanticError("Only Integer in [] in Subscript supported for now in annotation",
+                    if (ASR::is_a<ASR::ConstantInteger_t>(*value)) {
+                        ASR::ttype_t *itype = ASRUtils::TYPE(ASR::make_Integer_t(al, loc,
+                                4, nullptr, 0));
+                        dim.m_start = ASR::down_cast<ASR::expr_t>(ASR::make_ConstantInteger_t(al, loc, 1, itype));
+                        dim.m_end = value;
+                    } else if (ASR::is_a<ASR::Var_t>(*value)) {
+                        ASR::ttype_t *itype = ASRUtils::TYPE(ASR::make_Integer_t(al, loc,
+                                4, nullptr, 0));
+                        dim.m_start = ASR::down_cast<ASR::expr_t>(ASR::make_ConstantInteger_t(al, loc, 1, itype));
+                        dim.m_end = value;
```

Comment:
```suggestion
                    if (ASR::is_a<ASR::ConstantInteger_t>(*value) || ASR::is_a<ASR::Var_t>(*value)) {
                        ASR::ttype_t *itype = ASRUtils::TYPE(ASR::make_Integer_t(al, loc,
                                4, nullptr, 0));
                        dim.m_start = ASR::down_cast<ASR::expr_t>(ASR::make_ConstantInteger_t(al, loc, 1, itype));
                        dim.m_end = value;
```

Reply:
Thanks. It seems this indeed can be done, but I am waiting until I get the LLVM backend working, there is some bug somewhere, as it works in LFortran but not LPython.